content/docs/guides/dockerswarm.md

```diff
@@ -0,0 +1,227 @@
+---
+title: Docker Swarm
+sort_rank: 1
+---
+
+# Docker Swarm
+
+Prometheus can discover targets in a [Docker Swarm][swarm] cluster, as of
+v2.20.0. This guide demonstrates how to use that service discovery mechanism.
+
+## Docker Swarm service discovery architecture
+
+The [Docker Swarm service discovery][swarmsd] contains 3 different roles: nodes, services,
+and tasks.
+
+The first role, **nodes**, represents the hosts that are part of the Swarm. It
+can be used to automatically monitor the Docker daemons or the Node Exporters
+who run on the Swarm hosts.
+
+The second one, **services**, will discover the services deployed in the
+swarm. This could be used if you only want to scrape one of the targets at the
+same time, as it would used the load balanced IP address. Each port will be
+discovered individually. Services with multiple ports would be mapped to
+multiple targets in Prometheus, and service that do not expose ports won't be
+discovered.
+
+The third role, **tasks**, represents any individual container deployed in the
+swarm. Each service is be mapped by one or multiple containers.
+
+**Note**: The rest of this post assumes that you have a Swarm running.
+
+## Monitoring Docker daemons
+
+Let's dive into the service discovery itself.
+
+Docker itself, as a daemon, exposes [metrics][dockermetrics] that can be
+ingested by a Prometheus server.
+
+You can enable them by editing `/etc/docker/daemon.json` and setting the
+following properties:
+
+```json
+{
+  "metrics-addr" : "0.0.0.0:9323",
+  "experimental" : true
+}
+```
+
+Instead of `0.0.0.0`, you can set the IP of the Docker Swarm node.
+
+A restart of the daemon is required to take the new configuration into account.
+
+The [Docker documentation][dockermetrics] contains more info about this.
+
+Then, you can configure Prometheus to scrape the Docker daemon, by providing the
+following `prometheus.yml` file:
+
+
+```yaml
+scrape_configs:
+  # Make Prometheus scrape itself for metrics.
+  - job_name: 'prometheus'
+    static_configs:
+    - targets: ['localhost:9090']
+
+  # Create a job for Docker daemons.
+  - job_name: 'docker'
+    dockerswarm_sd_configs:
+      - host: unix:///var/run/docker.sock
```

Comment:
You didn't mention anything about the management node, so this will not work.

Reply:
I have added a new section

Reply:
I'm not seeing anything about management nodes, did you forget to push something?

Reply:
https://github.com/prometheus/docs/pull/1696/files#diff-1fc2f14b954f6c9cf7c6e6c7a95b05fcR35